src/worker-runner.js

```diff
@@ -32,6 +32,27 @@ export default async function ({
 }) {
   port.postMessage("start");
 
+  const {setupFiles, snapshotSerializers} = test.context.config;
+
+  // https://github.com/facebook/jest/issues/11038
+  for (const setupFile of setupFiles) {
+    const setup = await import(pathToFileURL(setupFile));
+
+    if (setup && setup.default) {
+      await setup.default;
+    }
+  }
+
+  for (const snapshotSerializer of snapshotSerializers) {
+    let serializer = await import(pathToFileURL(snapshotSerializer));
+
+    if (serializer && serializer.default) {
+      serializer = serializer.default;
+    }
```

Comment:
This should be enough, since the serializer is always stored on `module.exports` and thus becomes the default export:
```js
const { default: serializer } = await import(pathToFileURL(snapshotSerializer));
```

Also, probably we should add them in reverse order: ```suggestion
    if (serializer.default) {
```
I think that `serializer` will always be an object, because it's `import()`'s result.

Also, we should probably add the serializers in reverse order: https://github.com/facebook/jest/blob/a20bd2c31e126fc998c2407cfc6c1ecf39ead709/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts#L149-L152